libcxx/include/__atomic/atomic_sync.h

```diff
@@ -58,6 +62,7 @@ struct __atomic_waitable< _Tp,
 #if _LIBCPP_STD_VER >= 20
 #  if _LIBCPP_HAS_THREADS
 
+// old dylib interface kept for backwards compatibility
```

Comment:
Let's do

```c++
#if !_LIBCPP_AVAILABILITY_HAS_NEW_SYNC
<old declarations>
#endif
```

That way we get a static guarantee that we don't try calling these things when we're not supposed to.